dids/src/main/kotlin/web5/sdk/dids/methods/dht/DidDht.kt

```diff
@@ -45,393 +79,500 @@ public class DidDhtConfiguration internal constructor(
+    val dnsPacket = toDnsPacket(didDocument, types)
+    val bep44Message = Dht.createBep44PutRequest(manager, getIdentityKid(didDocument), dnsPacket)
+    dht.pkarrPut(publishId, bep44Message)
+  }
 
-      // map key ids to their verification method ids
-      val verificationMethodsById = mutableMapOf<String, String>()
+  /**
+   * Returns the suffix of the DID, which is the last part of the DID's method-specific identifier.
+   *
+   * @param id The DID to get the suffix of.
+   * @return The suffix of the DID [String].
+   */
+  public fun suffix(id: String): String {
+    return id.split(":").last()
+  }
 
-      // track all verification methods and services by their ids
-      val verificationMethodIds = mutableListOf<String>()
-      val serviceIds = mutableListOf<String>()
+  /**
+   * Returns the kid of the identity key for a did:dht DID Document.
+   *
+   * @param didDocument The DID Document to get the kid of.
+   * @return The kid of the identity key.
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   */
+  public fun getIdentityKid(didDocument: DIDDocument): String {
+    validate(didDocument.id.toString())
 
-      // Add Resource Records for each Verification Method
-      didDocument.verificationMethods?.forEachIndexed { i, verificationMethod ->
-        val publicKeyJwk = JWK.parse(verificationMethod.publicKeyJwk)
-        val publicKeyBytes = Crypto.publicKeyToBytes(publicKeyJwk)
-        val base64UrlEncodedKey = Base64URL(Convert(publicKeyBytes).toBase64Url(padding = false))
-        val vmId = "k$i"
+    val publicKeyJwk = JWK.parse(didDocument.verificationMethods?.first()?.publicKeyJwk)
+    return publicKeyJwk.keyID
+  }
 
-        verificationMethodsById[verificationMethod.id.toString()] = vmId
+  /**
+   * Generates the identifier for a did:dht DID given its identity key.
+   *
+   * @param identityKey the key used to generate the DID's identifier
+   */
+  public fun getDidIdentifier(identityKey: JWK): String {
+    val publicKeyJwk = identityKey.toPublicJWK()
+    val publicKeyBytes = Crypto.publicKeyToBytes(publicKeyJwk)
+    val zBase32Encoded = ZBase32.encode(publicKeyBytes)
+    return "did:dht:$zBase32Encoded"
+  }
 
-        val keyType = when (publicKeyJwk.algorithm) {
-          JWSAlgorithm.EdDSA -> 0
-          JWSAlgorithm.ES256K -> 1
-          else -> throw IllegalArgumentException("unsupported algorithm: ${publicKeyJwk.algorithm}")
-        }
+  /**
+   * Checks whether a given DID identifier conforms to the "did:dht" method. This checks that the DID starts with
+   * "did:dht" and that the suffix is a valid z-base-32 encoded public key.
+   *
+   * @param did The DID to check.
+   * @throws IllegalArgumentException if the provided DID does not conform to the "did:dht" method.
+   * @throws ParserException if the provided DID is not a valid DID.
+   */
+  public fun validate(did: String) {
```

Comment:
this is only useful if public...

Reply:
Similar to above, are there callers with this use case? If not, I'd suggest removing. 

If you decide to keep, can you make this part of the `DidMethod` interface, so that all methods implement it in a consistent manner?

Reply:
changed to internal